### ISSUE TITLE
Refactoring

### ISSUE THREAD

**Issue by YoshiyukiKato** [YoshiyukiKato/grpc-mock#20]
Because this lib's code has been complex and dirty.

**Comment by celesteking** [comment]
Yes, it is. your code is very hard to read. You should read a book on programming styles and another one on patterns.
